packages/functional/tests/ui/mobile/checkout-experience.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { faker } from '@faker-js/faker';
+import { expect, test } from '@playwright/test';
+
+import { ProductActions } from '../../../actions/product-actions';
+import { CheckoutPage } from '../../../pages/checkout-page';
+
+const sampleProduct = '[Sample] Able Brewing System';
+const testUser = faker.person.firstName();
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/');
+  await page.getByLabel('Toggle navigation').click();
+  await page.getByLabel('Main').getByRole('link', { name: 'Shop All' }).click();
+
+  await expect(
+    page.getByRole('link', { name: '[Sample] Able Brewing System' }).first(),
+  ).toBeVisible();
+});
+
+/**
+ * @todo skipping this test as we need to setup a new gh action for ios tests that run on mac config
+ *
+ */
+test.skip('Checkout experience on ios mobile', async ({ page }) => {
+  await ProductActions.addProductToCart(page, sampleProduct);
+
+  await page.getByRole('link', { name: 'Cart Items 1' }).click();
+  await expect(page.getByRole('heading', { level: 1, name: 'Your cart' })).toBeVisible();
+
+  await page.getByRole('link', { name: 'Proceed to checkout' }).click();
+
+  await page.getByText('Email').fill(faker.internet.email());
+  await page.locator(CheckoutPage.CONTINUE_AS_GUEST).click();
+  await page.getByText('First Name').fill(testUser);
+  await page.getByText('Last Name').fill(faker.person.lastName());
+  await page.getByText('Company Name (Optional)').fill('BigCommerce');
+  await page.getByText('Phone Number (Optional)').fill(faker.phone.number());
+  await page.getByText('Address', { exact: true }).fill(faker.location.buildingNumber());
+  await page.getByText('City').fill(faker.location.city());
+  await page.locator(CheckoutPage.PROVINCE_CODE_SELECT).selectOption('MA');
+  await page.getByText('Postal Code').fill('01762');
+  await expect(page.locator('#checkout-shipping-continue')).toContainText('Continue');
+
+  await page.getByRole('button', { name: 'Continue' }).click();
+
+  await expect(page.getByRole('heading', { name: 'Payment', exact: true })).toBeVisible();
+
+  await page
+    .frameLocator(CheckoutPage.BIGPAY_CC_NUMBER_IFRAME)
```

Comment:
I'm curious if we can remove the `frameLocaltor` here? I assume not, but would love to see if it's possible.

Reply:
Tried this but it wasn't able to locate the elements without the frameLocator.